libs/validator/address_validator.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_ADDRESS_VALIDATOR_HPP
+#define IROHA_ADDRESS_VALIDATOR_HPP
+
+#include <regex>
+#include <string>
+
+namespace iroha {
+  namespace validator {
+
+  /**
+   * validates ip v4 address ending with port number
+   * @param address
+   * @return true if address is valid
+   */
+    bool is_valid_ipv4(const std::string &address) {
+      std::regex valid_ipv4(
+          R"((^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])):(6553[0-5]|655[0-2]\d|65[0-4]\d\d|6[0-4]\d{3}|[1-5]\d{4}|[1-9]\d{0,3}|0)$))");
```

Comment:
Imo this line should be splitted? It also would be nice if it is splitted semantically, e.g (note there's no additional `;` and `\` here):
```cpp
R"#^((([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}#"
  "([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])):"
R"#(6553[0-5]|655[0-2]\d|65[0-4]\d\d|6[0-4]\d{3}|[1-5]\d{4}|[1-9]\d{0,3}|0)$#";
```
Also don't use the raw string separator that is inside the string, it's harder to read

Reply:
I agree with separation. But sometimes it is needed to put raw string iterator in the middle.